src/include/duckdb/common/progress_bar.hpp

```diff
@@ -0,0 +1,57 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/common/progress_bar.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <iostream>
+#include <thread>
+#include <future>
+#include "duckdb.h"
+#include "duckdb/execution/executor.hpp"
+
+namespace duckdb {
+
+class ProgressBar {
+public:
+	explicit ProgressBar(Executor *executor, idx_t show_progress_after, idx_t time_update_bar, bool test = false)
+	    : executor(executor), show_progress_after(show_progress_after), time_update_bar(time_update_bar),
```

Comment:
To add to that, I would also like there to be a method on the ClientContext that exposes this similar to the query profiler, e.g. `double ClientContext::GetProgress()`. This should always return 0 if the progress bar is disabled.

This can also be used in the tests of the progress bar.